diff --git a/libos/test/fs/manifest.template b/libos/test/fs/manifest.template
index 612873c0..b7a33db1 100644
--- a/libos/test/fs/manifest.template
+++ b/libos/test/fs/manifest.template
@@ -10,6 +10,7 @@ fs.mounts = [
   { path = "/usr/{{ arch_libdir }}", uri = "file:/usr/{{ arch_libdir }}" },
   { path = "/mounted", uri = "file:tmp" },
 
+  { type = "encrypted", path = "/protected", uri = "file:protected" , key_name = "_sgx_mrenclave"},
   { type = "encrypted", path = "/tmp/enc_input", uri = "file:tmp/enc_input" },
   { type = "encrypted", path = "/tmp/enc_output", uri = "file:tmp/enc_output" },
   { type = "encrypted", path = "/mounted/enc_input", uri = "file:tmp/enc_input" },
diff --git a/libos/test/fs/meson.build b/libos/test/fs/meson.build
index 02c68830..c17c1e6e 100644
--- a/libos/test/fs/meson.build
+++ b/libos/test/fs/meson.build
@@ -44,6 +44,7 @@ tests = {
     'open_flags': {},
     'read_append': {},
     'read_write': {},
+    'rename_pf': {},
     'read_write_mmap': {},
     'seek_tell': {},
     'seek_tell_truncate': {},
diff --git a/libos/test/fs/rename_pf.c b/libos/test/fs/rename_pf.c
new file mode 100644
index 00000000..5439841d
--- /dev/null
+++ b/libos/test/fs/rename_pf.c
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define FILENAME "protected/data"
+
+int main(void) {
+    FILE* f;
+
+    if ((f = fopen(FILENAME, "wb")) == NULL) {
+        perror("first create failed");
+        exit(1);
+    }
+    if (fwrite("data", 1, 4, f) != 4) {
+        perror("first write failed");
+        fclose(f);
+        exit(1);
+    }
+    fclose(f);
+
+    rename(FILENAME, FILENAME ".bak");
+    printf("TEST OK\n");
+    return 0;
+}
+
diff --git a/libos/test/fs/test_enc.py b/libos/test/fs/test_enc.py
index 3b83a338..74fca8d0 100644
--- a/libos/test/fs/test_enc.py
+++ b/libos/test/fs/test_enc.py
@@ -81,6 +81,12 @@ class TC_50_EncryptedFiles(test_fs.TC_00_FileSystem):
             self.__decrypt_file(self.OUTPUT_FILES[i], dec_path)
             self.assertTrue(filecmp.cmp(self.INPUT_FILES[i], dec_path, shallow=False))
 
+    @unittest.skipIf(os.environ.get('SGX') != '1', 'Checking renaming of protected files for SGX only')
+    def test_099_rename_pf(self):
+        os.makedirs('protected', exist_ok=True)
+        stdout, stderr = self.run_binary(['rename_pf'])
+        self.assertIn('TEST OK', stdout)
+
     # overrides TC_00_FileSystem to change input dir (from plaintext to encrypted)
     def test_100_open_close(self):
         input_path = self.ENCRYPTED_FILES[-1] # existing file
diff --git a/libos/test/fs/tests.toml b/libos/test/fs/tests.toml
index 2e982fdb..9e70ea88 100644
--- a/libos/test/fs/tests.toml
+++ b/libos/test/fs/tests.toml
@@ -15,6 +15,7 @@ manifests = [
   "open_flags",
   "read_append",
   "read_write",
+  "rename_pf",
   "read_write_mmap",
   "seek_tell",
   "seek_tell_truncate",
