diff --git a/libos/test/fs/manifest.template b/libos/test/fs/manifest.template
index 3470be82..5c5c8150 100644
--- a/libos/test/fs/manifest.template
+++ b/libos/test/fs/manifest.template
@@ -10,7 +10,7 @@ fs.mounts = [
   { path = "{{ arch_libdir }}", uri = "file:{{ arch_libdir }}" },
   { path = "/usr/{{ arch_libdir }}", uri = "file:/usr/{{ arch_libdir }}" },
   { path = "/mounted", uri = "file:tmp" },
-
+  { type = "encrypted", path = "/protected", uri = "file:protected" , key_name = "_sgx_mrenclave"},
   { type = "encrypted", path = "/tmp/enc_input", uri = "file:tmp/enc_input" },
   { type = "encrypted", path = "/tmp/enc_output", uri = "file:tmp/enc_output" },
   { type = "encrypted", path = "/mounted/enc_input", uri = "file:tmp/enc_input" },
diff --git a/libos/test/fs/meson.build b/libos/test/fs/meson.build
index ce3f8b2e..7d587dde 100644
--- a/libos/test/fs/meson.build
+++ b/libos/test/fs/meson.build
@@ -43,6 +43,7 @@ tests = {
     'open_close': {},
     'open_flags': {},
     'read_write': {},
+    'rename_pf': {},
     'read_write_mmap': {},
     'seek_tell': {},
     'seek_tell_truncate': {},
diff --git a/libos/test/fs/rename_pf.c b/libos/test/fs/rename_pf.c
new file mode 100644
index 00000000..62e52f6f
--- /dev/null
+++ b/libos/test/fs/rename_pf.c
@@ -0,0 +1,22 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define FILENAME "protected/data"
+
+int main(void) {
+    FILE* f;
+
+    if ((f = fopen(FILENAME, "wb")) == NULL) {
+        perror("first create failed");
+        exit(1);
+    }
+    if (fwrite("data", 1, 4, f) != 4) {
+        perror("first write failed");
+        fclose(f);
+        exit(1);
+    }
+    fclose(f);
+
+    rename(FILENAME, FILENAME ".bak");
+    printf("TEST OK\n");
+    return 0;
+}
diff --git a/libos/test/fs/test_enc.py b/libos/test/fs/test_enc.py
index ee90890a..f53006cc 100644
--- a/libos/test/fs/test_enc.py
+++ b/libos/test/fs/test_enc.py
@@ -81,6 +81,12 @@ class TC_50_EncryptedFiles(test_fs.TC_00_FileSystem):
             self.__decrypt_file(self.OUTPUT_FILES[i], dec_path)
             self.assertTrue(filecmp.cmp(self.INPUT_FILES[i], dec_path, shallow=False))
 
+    @unittest.skipIf(os.environ.get('SGX') != '1', 'Checking renaming of protected files for SGX only')
+    def test_099_rename_pf(self):
+        os.makedirs('protected', exist_ok=True)
+        stdout, stderr = self.run_binary(['rename_pf'])
+        self.assertIn('TEST OK', stdout)
+
     # overrides TC_00_FileSystem to change input dir (from plaintext to encrypted)
     def test_100_open_close(self):
         input_path = self.ENCRYPTED_FILES[-1] # existing file
diff --git a/libos/test/fs/tests.toml b/libos/test/fs/tests.toml
index 362b7373..cd6d5ee4 100644
--- a/libos/test/fs/tests.toml
+++ b/libos/test/fs/tests.toml
@@ -14,6 +14,7 @@ manifests = [
   "open_close",
   "open_flags",
   "read_write",
+  "rename_pf",
   "read_write_mmap",
   "seek_tell",
   "seek_tell_truncate",
